pkg/custmsg/custom_message.go

```diff
@@ -11,17 +11,31 @@ import (
 	"github.com/smartcontractkit/chainlink-common/pkg/values"
 )
 
-type Labeler struct {
+type MessageEmitter interface {
+	// Emit sends a message to the labeler's destination.
+	Emit(string) error
+
+	// WithMapLabels sets the labels for the message to be emitted.  Labels are cumulative.
+	WithMapLabels(map[string]string) MessageEmitter
+
+	// With adds multiple key-value pairs to the emission.
+	With(keyValues ...string) MessageEmitter
+
+	// Labels returns a view of the current labels.
+	Labels() map[string]string
+}
+
+type labeler struct {
 	labels map[string]string
 }
 
-func NewLabeler() Labeler {
-	return Labeler{labels: make(map[string]string)}
+func NewLabeler() labeler {
+	return labeler{labels: make(map[string]string)}
 }
 
 // WithMapLabels adds multiple key-value pairs to the CustomMessageLabeler for transmission
 // With SendLogAsCustomMessage
-func (l Labeler) WithMapLabels(labels map[string]string) Labeler {
+func (l labeler) WithMapLabels(labels map[string]string) MessageEmitter {
```

Comment:
idiomatic go to accept interfaces as params and return structs. could we do that here?

Reply:
not a hard axiom.  In this case there is an exception for two reasons:
1. `labeler` is the only implementation of `MessageEmitter` (see Effective Go on [generality](https://go.dev/doc/effective_go#generality) of the interface rule)
2. the interface is self-referential, if the methods don't return `MessageEmitter` then `labeler` won't implement `MessageEmitter`

Reply:
`Labeler` was originally public, but all of its fields are private, which makes it redundant to the interface so I made it private.  That's a third reason to not return `labeler` directly.

Reply:
Let's keep `Labeler` public for continued direct consumption by the engine and registry syncer in core.